umap/static/umap/js/modules/schema.js

```diff
@@ -1,45 +1,52 @@
 import { translate } from './i18n.js'
 
+// Possible impacts
+// ['ui', 'data', 'limit-bounds', 'datalayer-index', 'remote-data', 'background']
+
 export const SCHEMA = {
-  zoom: {
-    type: Number,
-  },
-  scrollWheelZoom: {
+  browsable: {
     type: Boolean,
-    label: translate('Allow scroll wheel zoom?'),
+    impacts: ['ui'],
   },
-  scaleControl: {
+  captionBar: {
+    impacts: ['ui'],
     type: Boolean,
-    label: translate('Do you want to display the scale control?'),
-    default: true,
+    label: translate('Do you want to display a caption bar?'),
+    default: false,
+    impacts: [],
   },
-  moreControl: {
+  captionMenus: {
+    impacts: ['ui'],
     type: Boolean,
-    label: translate('Do you want to display the «more» control?'),
+    label: translate('Do you want to display caption menus?'),
     default: true,
   },
-  miniMap: {
-    type: Boolean,
-    label: translate('Do you want to display a minimap?'),
-    default: false,
-  },
-  displayPopupFooter: {
-    type: Boolean,
-    label: translate('Do you want to display popup footer?'),
-    default: false,
+  color: {
```

Comment:
I suggest to add a dedicated `test_on_edit.py` file with some basic test (at least one per "impact"), that we can then complete later with some edge cases.